src/autogluon/cloud/cluster/ray_aws_cluster_config_generator.py

```diff
@@ -0,0 +1,146 @@
+            To learn more,
+                https://docs.ray.io/en/latest/cluster/vms/references/ray-cluster-configuration.html#node-config
+        instance_count: int, default = None
+            Number of instance the cluster will launch.
+            If provided, will overwrite `available_node_types.min_workers` and `max_workers`
+            min_workers and max_workers will both equal to `instance_count` - 1 because there will be a head node.
+            This setting doesn't work when there's more than one definition of worker nodes.
+            To learn more,
+                https://docs.ray.io/en/latest/cluster/vms/references/ray-cluster-configuration.html#available-node-types-node-type-name-node-type-min-workers
+                https://docs.ray.io/en/latest/cluster/vms/references/ray-cluster-configuration.html#max-workers
+                https://docs.ray.io/en/latest/cluster/vms/references/ray-cluster-configuration.html#cluster-configuration-max-workers
+        worker_node_name: str, default = worker
+            Name of the worker node inside the yaml file. This is used to correctly configure the instance count if specified.
+        volumes_size: int, default = None
+            Size in GB of the EBS volume to use for each of the node.
+            If provided, will overwrite `available_node_types.node_config.BlockDeviceMappings.Ebs.VolmueSize`
+            To learn more,
+                https://docs.ray.io/en/latest/cluster/vms/references/ray-cluster-configuration.html#available-node-types
+        custom_image_uri: str, default = None
+            Custom image to be used by the cluster container. The image MUST have Ray and AG installed.
+            If provided, will overwrite `docker.image`
+            To learn more,
+                https://docs.ray.io/en/latest/cluster/vms/references/ray-cluster-configuration.html#docker-image
+        """
+        self._update_instance_type(instance_type=instance_type)
+        self._update_instance_count(instance_count=instance_count, worker_node_name=worker_node_name)
+        self._update_volumn_size(volumes_size=volumes_size)
+        self._update_custom_image(custom_image_uri=custom_image_uri)
+
+    def _set_available_node_types(self):
+        """Set available node types to be default ones if user didn't provide any"""
+        default_config = self._default_config
+        available_node_types: Dict[str, Any] = self.config.get(AVAILABLE_NODE_TYPES, None)
+        if available_node_types is None:
+            available_node_types = default_config[AVAILABLE_NODE_TYPES]
+        self.config.update(available_node_types)
+
+    def _set_provider(self):
+        """Set provider to be default ones if user didn't provide any"""
+        default_config = self._default_config
+        provider: Dict[str, Any] = self.config.get(PROVIDER, None)
+        if provider is None:
+            provider = default_config[PROVIDER]
+        self.config.update({PROVIDER: provider})
```

Comment:
Can we templatize repeating methods: I see lots of repeating code which can be reuse between the `_update_xxx` methods.

Reply:
It is kind of hard to templatize these methods because they are essentially trying to update keys and values lying in different layers of the yaml file. The method looks similar, but because the structure of the yaml file differs for each key, it's actually specific